File: 2020/01/report_repair2.py
# report_repair.py
# Callum Howard, 2020

import sys
from itertools import combinations

def solve2(nums, target):
    for a in nums:
        for b in nums:
            if target - a - b in nums:
                return (a * b * (target - a - b))

def solve(nums, target):
    return next(a * b * (target - a - b) for a, b in combinations(nums, 2) if target - a - b in nums)

print(solve(set([int(n) for n in sys.stdin.readlines()]), 2020))

File: 2020/01/report_repair2.go
// Report Repair
// https://adventofcode.com/2020/day/1

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func solve(nums map[int]bool, target int) int {
	for keyA := range nums {
		for keyB := range nums {
			if nums[target-keyA-keyB] {
				return keyA * keyB * (target - keyA - keyB)
			}
		}
	}
	return 0
}

func main() {
	const target = 2020
	var scanner = bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanLines)

	var nums = make(map[int]bool)
	for scanner.Scan() {
		var num, _ = strconv.Atoi(scanner.Text())
		nums[num] = true
	}

	fmt.Println(solve(nums, target))
}
